spec/models/order_spec.rb

```diff
@@ -1,5 +1,49 @@
 require 'rails_helper'
 
 RSpec.describe Order, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:order){
+    FactoryBot.create :order
+  }
+
+  describe "delegations" do
+    it { should delegate_method(:name).to(:user) }
+  end
+
+  describe "enums" do
+    it { should define_enum_for(:status) }
+  end
+
+  describe "associations" do
+    it { should belong_to(:user) }
+    it { should have_many(:order_details).dependent(:destroy)}
+    it { should have_many(:products).through(:order_details) }
+  end
+
+  describe "check scope" do
+    it "created in this month" do
```

Comment:
phân biệt describe, context, it

Reply:
1. Describe:
- Describe dùng để nhóm các test case theo class, module, method hoặc theo action của controller.
- Cần có mô tả cho trường hợp không là method.
- Trong trường hợp đối với instance method thì gắn thêm "#" ví dụ "#method".
- Trong trường hợp đối với class method thì gắn thêm "." ví dụ ".method".
2. Context:
- Context dùng để nhóm các điều kiện của example.
- Phần mô tả của context thường bắt đầu bằng when hoăc with ex:
3. It:
- It tương ứng với 1 test case.

Chúng ta có thể mô tả cho it để test case được rõ ràng